### PR TITLE
Bump up tiledb version to 2.21.1

### PR DESCRIPTION
**Pull request by dudoslav**
Bump up:

TileDB-Py version to 0.27.0
TileDB version to 2.21.1

NOTE: I had to modify `DownloadPrebuiltTileDB.cmake` since we are still using superbuild. Also source distribution needs its own logic added.

This should be release: `2.0`

### CODE REVIEW

**Review by ihnorton — APPROVED** [review]
(FWIW I thought we were going to pull the `DownloadPrebuildTiledb.cmake` from the release bundle, but up to you if you prefer to vendor it)

File: src/include/test/unit_api_ivf_flat_index.cc
Comment (ihnorton) [inline src/include/test/unit_api_ivf_flat_index.cc:285]: @lums658 @jparismorgan please confirm if this is ok. The rest of the higher-precision comparisons in this file are like this. We saw a failure here in [this run](https://github.com/TileDB-Inc/TileDB-Vector-Search/actions/runs/8438568931/job/23111048191?pr=291):
```
9: /home/runner/work/TileDB-Vector-Search/TileDB-Vector-Search/src/include/test/unit_api_ivf_flat_index.cc:285: FAILED:
9:   CHECK( recall == 1.0 )
9: with expansion:
9:   0.999 == 1.0
9: with message:
```

**Review by ihnorton — CHANGES_REQUESTED** [review]
Needs approval of the CI-fix commit I just added, before merging.

**Review by jparismorgan — APPROVED** [review]

**Review by ihnorton — APPROVED** [review]

**Pull request merged**
